.circleci/golden/expected/src/OpenAPI/Types/Cat.hs

```diff
@@ -0,0 +1,94 @@
+import qualified Data.ByteString.Char8
+import qualified Data.ByteString.Char8 as Data.ByteString.Internal
+import qualified Data.Functor
+import qualified Data.Scientific
+import qualified Data.Text
+import qualified Data.Text.Internal
+import qualified Data.Time.Calendar as Data.Time.Calendar.Days
+import qualified Data.Time.LocalTime as Data.Time.LocalTime.Internal.ZonedTime
+import qualified GHC.Base
+import qualified GHC.Classes
+import qualified GHC.Int
+import qualified GHC.Show
+import qualified GHC.Types
+import qualified OpenAPI.Common
+import OpenAPI.TypeAlias
+import {-# SOURCE #-} OpenAPI.Types.PetByType
+
+-- | Defines the data type for the schema Cat
+-- 
+-- 
+data Cat = Cat {
+  -- | age
+  catAge :: (GHC.Maybe.Maybe GHC.Types.Int)
+  -- | ananyoftype
+  , catAnanyoftype :: (GHC.Maybe.Maybe Data.Aeson.Types.Internal.Object)
+  -- | another_relative
+  , catAnother_relative :: (GHC.Maybe.Maybe CatAnother_relativeVariants)
+  -- | hunts
+  , catHunts :: (GHC.Maybe.Maybe GHC.Types.Bool)
+  -- | relative
+  , catRelative :: (GHC.Maybe.Maybe CatRelativeVariants)
+  } deriving (GHC.Show.Show
+  , GHC.Classes.Eq)
+instance Data.Aeson.Types.ToJSON.ToJSON Cat
+    where toJSON obj = Data.Aeson.object ((Data.Aeson..=) "age" (catAge obj) : (Data.Aeson..=) "ananyoftype" (catAnanyoftype obj) : (Data.Aeson..=) "another_relative" (catAnother_relative obj) : (Data.Aeson..=) "hunts" (catHunts obj) : (Data.Aeson..=) "relative" (catRelative obj) : [])
+          toEncoding obj = Data.Aeson.pairs ((Data.Aeson..=) "age" (catAge obj) GHC.Base.<> ((Data.Aeson..=) "ananyoftype" (catAnanyoftype obj) GHC.Base.<> ((Data.Aeson..=) "another_relative" (catAnother_relative obj) GHC.Base.<> ((Data.Aeson..=) "hunts" (catHunts obj) GHC.Base.<> (Data.Aeson..=) "relative" (catRelative obj)))))
+instance Data.Aeson.Types.FromJSON.FromJSON Cat
+    where parseJSON = Data.Aeson.Types.FromJSON.withObject "Cat" (\obj -> ((((GHC.Base.pure Cat GHC.Base.<*> (obj Data.Aeson.Types.FromJSON..:? "age")) GHC.Base.<*> (obj Data.Aeson.Types.FromJSON..:? "ananyoftype")) GHC.Base.<*> (obj Data.Aeson.Types.FromJSON..:? "another_relative")) GHC.Base.<*> (obj Data.Aeson.Types.FromJSON..:? "hunts")) GHC.Base.<*> (obj Data.Aeson.Types.FromJSON..:? "relative"))
+-- | Define the one-of schema CatAnother_relative
+-- 
+-- 
+data CatAnother_relativeVariants
+    = CatAnother_relativeCat Cat
+    | CatAnother_relativePetByType PetByType
+    | CatAnother_relativeText Data.Text.Internal.Text
+    deriving (GHC.Show.Show, GHC.Classes.Eq)
+instance Data.Aeson.Types.ToJSON.ToJSON CatAnother_relativeVariants
+    where toJSON (CatAnother_relativeCat a) = Data.Aeson.Types.ToJSON.toJSON a
+          toJSON (CatAnother_relativePetByType a) = Data.Aeson.Types.ToJSON.toJSON a
+          toJSON (CatAnother_relativeText a) = Data.Aeson.Types.ToJSON.toJSON a
+instance Data.Aeson.Types.FromJSON.FromJSON CatAnother_relativeVariants
+    where parseJSON val = case Data.Aeson.Types.FromJSON.fromJSON val of
```

Comment:
Wouldn't using `<|>` be much simpler than multiple levels of these cases?

Reply:
Yes, this would be a nice improvement! Thank you for the tip ;-)